gateway/reverse_proxy.go

```diff
@@ -419,15 +419,20 @@ func (p *ReverseProxy) defaultTransport(dialerTimeout float64) *http.Transport {
 }
 
 func singleJoiningSlash(a, b string, disableStripSlash bool) string {
-	if disableStripSlash && (len(b) == 0 || b == "/") {
-		return a
+	var clean = func(s string) string {
+		s = strings.ReplaceAll(s, "//", "/")
```

Comment:
why do we try to handle `//bar` case?

Reply:
Pre-existing in tests, it would have trimmed excess // from the path. Even if a=`/foo//` it would have trimmed the `//` with TrimRight.

Reply:
Removed. It was a requirement to keep `//` as part of path args, e.g. an urlencoded URL.